tests/test_modeling_xlnet.py

```diff
@@ -305,21 +305,21 @@ def create_and_check_xlnet_lm_head(
 
         result1 = model(input_ids_1, token_type_ids=segment_ids, labels=lm_labels)
 
-        result2 = model(input_ids_2, token_type_ids=segment_ids, labels=lm_labels, mems=result1["mems"])
+        result2 = model(input_ids_2, token_type_ids=segment_ids, labels=lm_labels, mems=result1.mems)
 
         _ = model(input_ids_q, perm_mask=perm_mask, target_mapping=target_mapping)
 
         self.parent.assertEqual(result1.loss.shape, ())
         self.parent.assertEqual(result1.logits.shape, (self.batch_size, self.seq_length, self.vocab_size))
         self.parent.assertListEqual(
-            list(list(mem.size()) for mem in result1["mems"]),
+            list(list(mem.size()) for mem in result1.mems),
             [[self.seq_length, self.batch_size, self.hidden_size]] * self.num_hidden_layers,
         )
 
         self.parent.assertEqual(result2.loss.shape, ())
         self.parent.assertEqual(result2.logits.shape, (self.batch_size, self.seq_length, self.vocab_size))
         self.parent.assertListEqual(
-            list(list(mem.size()) for mem in result2["mems"]),
+            list(list(mem.size()) for mem in result2.mems),
```

Comment:
can this be 
`self.assertListEqual([mem.size() for mem in result2.mems], [(self.seq_length, self.batch_size, self.hidden_size)]*self.num_hidden_layers)`

somewhat cleaner IMO but no big deal.

Reply:
plus `.size()` => `.shape` - will do. thank you.

Reply:
So changing:
```
-            list(list(mem.shape) for mem in mems_1),
-            [[self.mem_len, self.batch_size, self.hidden_size]] * self.num_hidden_layers,
+            [mem.shape for mem in mems_1],
+            [(self.mem_len, self.batch_size, self.hidden_size)] * self.num_hidden_layers,
```

Perl:

```
perl -0777 -pi -e 's#self.parent.assertListEqual\(
[\s\n]*
list \( list \( mem.(?:shape|size\(\))\)\sfor\smem\s(.*?) \),
[\s\n]*
\[\[ (.*?) \]\]\s\*\sself.num_hidden_layers,
#self.parent.assertListEqual(
            [mem.shape for mem $1],
            [($2)] * self.num_hidden_layers,#xmsg' tests/test*py
```